SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp

```diff
@@ -0,0 +1,351 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+ 
+#include "Utils/EntityFactory.h"
+ 
+#include "Engine.h"
+ 
+#include "EngineClasses/SpatialActorChannel.h"
+#include "EngineClasses/SpatialNetDriver.h"
+#include "EngineClasses/SpatialPackageMapClient.h"
+#include "Interop/SpatialRPCService.h"
+#include "Schema/AlwaysRelevant.h"
+#include "Schema/AuthorityIntent.h"
+#include "Schema/Heartbeat.h"
+#include "Schema/ClientRPCEndpointLegacy.h"
+#include "Schema/ServerRPCEndpointLegacy.h"
+#include "Schema/RPCPayload.h"
+#include "Schema/Singleton.h"
+#include "Schema/SpawnData.h"
+#include "Utils/ComponentFactory.h"
+#include "Utils/InterestFactory.h"
+#include "Utils/SpatialActorUtils.h"
+ 
+namespace SpatialGDK
+{
+ 
+EntityFactory::EntityFactory(USpatialNetDriver* InNetDriver, SpatialRPCService* InRPCService)
```

Comment:
@aleximprobable How does this fit in with your views of refactoring for the purpose of testing?

Reply:
I can't say that the suggested change is worse than the current version testing-wise. This PR just moves some code to a new file. To make it testable we will need some other refactoring. Maybe @JHuculak has some ideas, since he mentions `breaking it
down into a number of smaller, more easily parseable functions` in the future.

Reply:
I haven't dug too deeply into the organization or layout of the function; I'd have to sit down and take a bit of a more serious look at it before making any actionable suggestions. Off the top of my head, though:

1. Some kind of helper class to deal with all the ACL stuff might make it a bit more parseable and would likely be easier to test in isolation.
2. Group things by functionality (e.g., these are my networking components, these are my actor-specific components, etc.), by source (e.g., which components I use here depend on what I have set in `SpatialGDKSettings`, these ones depend on what Class flags the Actor has / whether it's a `PlayerController` / etc., these ones depend on what we read out of the Actor's `ClassInfo`), or some combination of the two. While this is more or less just reorganization rather than refactoring, breaking it into smaller chunks should make it easier to test.

Reply:
That sounds reasonable.